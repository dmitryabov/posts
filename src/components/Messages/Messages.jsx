import React from 'react';
import Message from './Message/Message';

const Messages = (props) => {
    return (
        <div>
          <h2>List of messages</h2>
           {props.messages.map( message => <Message key={message.id} message={message}/> )}
        </div>
    )
}

export default Messages;