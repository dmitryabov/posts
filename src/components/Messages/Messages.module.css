
.topnav .search_container {
  float: right;
}

.topnav input[type=text] {
  padding: 6px;
  margin-top: 8px;
  font-size: 17px;
  border: none;
}

@media screen and (max-width: 600px) {
  .topnav .search_container {
    float: none;
  }
  .topnav a, .topnav input[type=text], .topnav .search_container button {
    float: none;
    display: block;
    text-align: left;
    width: 100%;
    margin: 0;
    padding: 14px;
  }
  .topnav input[type=text] {
    border: 1px solid #ccc;  
  }
}
