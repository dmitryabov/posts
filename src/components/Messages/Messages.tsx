import React, {useState, useEffect} from 'react';
import style from './Messages.module.css';
import Message from './Message/Message';
import Preloader from '../common/Preloader/Preloader';
import {MessagesType} from '../../redux/messages-reduser';
import {UsersType} from '../../redux/users-reduser';


type PropsType = {
    messages: Array<MessagesType>
    isFetching: boolean
    users: Array<UsersType>
    user: UsersType
}

const Messages: React.FC<PropsType> = (props) => {
    let [messages, setMessages] = useState(props.messages);
    
    useEffect(() => {
        setMessages(props.messages);
    }, [props.messages]);

    const filterList = (e: any) => {
        let filteredList = props.messages.filter(function(message){
            return message.title.toLowerCase().search(e.target.value.toLowerCase())!== -1;
        })
        setMessages(filteredList);
    }
 
    return (
        <div>
          {props.isFetching ? <Preloader /> : null}
          <h2>List of messages</h2>
          <div className={style.search_container}> 
            <span>messages filter on title </span>
            <input placeholder="Search.." onChange={filterList} name="search"/>
          </div>
           {messages.map( message => <Message key={message.id + message.title} 
                                              message={message} 
                                              user={props.users.find(user => user.id === message.userId)}/> )}
        </div>
    )
}

export default Messages;

