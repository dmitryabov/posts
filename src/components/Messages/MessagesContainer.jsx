import React from 'react';
import Messages from './Messages';

 
class MessagesContainer extends React.Component {
   refreshProfile () {  
      this.props.getMessages();
      this.props.getUsers();
   }

   componentDidMount () {
      this.refreshProfile()
   } 
   
   render () { 
     return <Messages { ...this.props} />
   }
}






export default MessagesContainer;