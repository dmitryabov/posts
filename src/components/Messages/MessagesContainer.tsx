import React from 'react';
import Messages from './Messages';
import {MessagesType} from '../../redux/messages-reduser';
import {UsersType} from '../../redux/users-reduser';


type PropsType = {
   getMessages: () => void
   getUsers: () => void
   messages: Array<MessagesType>
   isFetching: boolean
   users: Array<UsersType>
   user: UsersType
}

 
class MessagesContainer extends React.Component<PropsType> {
   refreshProfile () {  
      this.props.getMessages();
      this.props.getUsers();
   }

   componentDidMount () {
      this.refreshProfile()
   } 
   
   render () { 
     return <Messages { ...this.props} />
   }
}






export default MessagesContainer;