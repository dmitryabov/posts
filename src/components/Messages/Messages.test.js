import React from "react";
import renderer from "react-test-renderer";
import Messages from "./Messages";


const props = {
  messages: [
    {
      "userId": 1,
      "id": 1,
      "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
      "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto"
    } 
  ],
  isFetching: true,
  users: [
    {
        "id": 1,
        "name": "Leanne Graham",
        "username": "Bret",
        "email": "Sincere@april.biz",
        "address": {
          "street": "Kulas Light",
          "suite": "Apt. 556",
          "city": "Gwenborough",
          "zipcode": "92998-3874",
          "geo": {
            "lat": "-37.3159",
            "lng": "81.1496"
          }
        },
        "phone": "1-770-736-8031 x56442",
        "website": "hildegard.org",
        "company": {
          "name": "Romaguera-Crona",
          "catchPhrase": "Multi-layered client-server neural-net",
          "bs": "harness real-time e-markets"
        }
      }
  ],
  user: {
    "id": 1,
    "name": "Leanne Graham",
    "username": "Bret",
    "email": "Sincere@april.biz",
    "address": {
      "street": "Kulas Light",
      "suite": "Apt. 556",
      "city": "Gwenborough",
      "zipcode": "92998-3874",
      "geo": {
        "lat": "-37.3159",
        "lng": "81.1496"
      }
    },
    "phone": "1-770-736-8031 x56442",
    "website": "hildegard.org",
    "company": {
      "name": "Romaguera-Crona",
      "catchPhrase": "Multi-layered client-server neural-net",
      "bs": "harness real-time e-markets"
    }
  }
}



it(`Render Messages`, () => {
  const tree = renderer
  .create(<Messages
    isFetching={true}
    {...props}
  />, {
    createNodeMock: ()=>{
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
