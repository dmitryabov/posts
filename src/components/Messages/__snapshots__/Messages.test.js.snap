// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Messages 1`] = `
<div>
  <div>
    <img
      alt="prelouder"
      src="spinner.svg"
    />
  </div>
  <h2>
    List of messages
  </h2>
  <div
    className="search_container"
  >
    <span>
      messages filter on title 
    </span>
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search.."
    />
  </div>
  <div
    className="container"
  >
    <p>
      <span>
        Leanne Graham
      </span>
       
      Bret
    </p>
    <p>
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </p>
    <p>
      <span>
        quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
      </span>
    </p>
  </div>
</div>
`;

exports[`Render User 1`] = `
<div>
  <div>
    <img
      alt="prelouder"
      src="spinner.svg"
    />
  </div>
  <h2>
    List of messages
  </h2>
  <div
    className="search_container"
  >
    <span>
      messages filter on title 
    </span>
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search.."
    />
  </div>
  <div
    className="container"
  >
    <p>
      <span>
        Leanne Graham
      </span>
       
      Bret
    </p>
    <p>
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </p>
    <p>
      <span>
        quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto
      </span>
    </p>
  </div>
</div>
`;
